src/tests/ftest/control/ms_resilience.py

```diff
@@ -0,0 +1,308 @@
+    """Test Class Description:
+    Verify that expected MS functionality is retained in various
+    failure/recovery scenarios.
+
+    The raft protocol guarantees 2N+1 resiliency, where N is the number of
+    nodes that can fail while leaving the cluster in a state where it can
+    continue to make progress. e.g. with N=1, a minimum of 3 nodes is needed
+    in the cluster in order to tolerate 1 node failure; with N=2, a minimum
+    of 5 nodes is required, etc.
+
+    :avocado: recursive
+    """
+
+    def __init__(self, *args, **kwargs):
+        """Inititialize a ManagementServiceResilience object."""
+        super().__init__(*args, **kwargs)
+        self.setup_start_servers = False
+        self.start_servers_once = False
+        self.L_QUERY_TIMER = 300 # adjust down as we learn more
+
+    def find_pool(self, uuid):
+        """Find a pool in the output of `dmg pool list`.
+
+        Args:
+            uuid (str): Pool UUID to find.
+
+        Returns:
+            Pool entry, if found, or None.
+        """
+        pools = self.get_dmg_command().pool_list()["response"]["pools"]
+        for pool in pools:
+            if pool["uuid"].lower() == uuid.lower():
+                return pool
+        return None
+
+    def create_pool(self):
+        """Create a pool on the server group."""
+        self.add_pool(create=False)
+        self.pool.name.value = self.server_group
+        self.log.info("*** creating pool")
+        self.pool.create()
+
+        self.log.info("Pool UUID %s on server group: %s",
+                        self.pool.uuid, self.server_group)
+        # Verify that the pool persisted.
+        if self.find_pool(self.pool.uuid):
+            self.log.info("Found pool in system.")
+        else:
+            self.fail("No pool found in system.")
+
+    def get_leader(self):
+        """Fetch the current system leader.
+
+        Returns:
+            str: hostname of the MS leader, or None
+        """
+        sys_leader_info = self.get_dmg_command().system_leader_query()
+        l_addr = sys_leader_info["response"]["CurrentLeader"]
+
+        if not l_addr:
+            return None
+
+        l_hostname, _, _ = socket.gethostbyaddr(l_addr.split(":")[0])
+        l_hostname = l_hostname.split(".")[0]
+
+        return l_hostname
+
+    def verify_leader(self, replicas):
+        """Verify the leader of the MS is in the replicas.
+
+        Args:
+            replicas (list): list of hostnames representing the access points
+                for the MS.
+
+        Returns:
+            str: address of the MS leader.
+
+        """
+        l_hostname = None
+        start = time.time()
+        while not l_hostname and (time.time() - start) < self.L_QUERY_TIMER:
+            l_hostname = self.get_leader()
+            # Check that the new leader is in the access list
+            if l_hostname not in replicas:
+                self.log.error("Selected leader <{}> is not within the replicas"
```

Comment:
(pylint-logging-format-interpolation) Use lazy % formatting in logging functions